pkg/leaderelection/leaderelection.go

```diff
@@ -48,7 +48,7 @@ func StartLeaderElection(ctx *context2.ControllerContext, scheme *runtime.Scheme
 
 	// Lock required for leader election
 	rl, err := resourcelock.New(
-		resourcelock.ConfigMapsLeasesResourceLock,
+		resourcelock.LeasesResourceLock,
```

Comment:
Based on [this](https://github.com/kubernetes/client-go/blob/4dd144a93b873d7597a1002ab06d763a3edb0ac3/tools/leaderelection/resourcelock/interface.go#L185)
``

Reply:
Does this require additional or changed rbac permissions?

Reply:
I tried to search more on maybe some migration procedures but couldn't find anything except that log.

Reply:
Yeah we need to double check that this isn't causing issues, also is this really completely removed?

Reply:
[This](url) is the commit where the removal has happened.

Reply:
I found more on this as below.

This [#117558](https://github.com/kubernetes/kubernetes/pull/117558) is the PR in kubernetes that introduces the current change and describes it. It all started in [#80289](https://github.com/kubernetes/kubernetes/issues/80289) with the intention to migrate leader election to use Lease API. The migration path was decided as
endpoints/configmap -> endpointsleases/configmapleases -> lease.

I followed our changes in vcluster where we went from [configmap](https://github.com/loft-sh/vcluster/commit/a6eead1f44d3eb956f90e4faf9f237972c2e9173) lock to [configmapleases](https://github.com/loft-sh/vcluster/commit/a0716b6bddd490121d9acac13c90eefd673b8f45#diff-2b56d264d4ca0994b896902728fb1a43c9524c2e0a39fdc9c10edb4447a4fa02) lock and now we are actually in the last phase of the migration i.e. to go from configmapleases to leases.

As for the RBAC, I see that the necessary rules for the "coordination.k8s.io"/leases were already added in [this](https://github.com/loft-sh/vcluster/commit/a0716b6bddd490121d9acac13c90eefd673b8f45#diff-2b56d264d4ca0994b896902728fb1a43c9524c2e0a39fdc9c10edb4447a4fa02) commit.